runtime/roots.c

```diff
@@ -42,14 +42,15 @@
 intnat caml_globals_inited = 0;
 #endif
 
-CAMLexport void (*caml_scan_roots_hook)
-  (scanning_action, void* fdata, caml_domain_state *) = NULL;
+CAMLexport _Atomic scan_roots_hook caml_scan_roots_hook = NULL;
```

Comment:
`camlatomic.h` does not assume that `stdatomic.h` to be present. See https://github.com/ocaml/ocaml/blob/trunk/runtime/caml/camlatomic.h. Is this breaking this assumption?